packages/x-components/src/views/Layout.vue

```diff
@@ -125,27 +125,26 @@
 
           <!-- Results -->
           <ResultsList v-infinite-scroll:body-scroll>
-            <template #layout="{ results }">
-              <BaseVariableColumnGrid
-                #default="{ item: result }"
-                :animation="resultsAnimation"
-                :items="results"
-              >
-                <article class="result" style="max-width: 300px">
-                  <BaseResultImage class="x-picture--colored" :result="result">
-                    <template #placeholder>
-                      <div style="padding-top: 100%; background-color: lightgray"></div>
-                    </template>
-                    <template #fallback>
-                      <div style="padding-top: 100%; background-color: lightsalmon"></div>
-                    </template>
-                  </BaseResultImage>
-                  <h1 class="x-title3">{{ result.name }}</h1>
-                </article>
-              </BaseVariableColumnGrid>
-            </template>
+            <BannersList>
+              <PromotedsList>
+                <BaseVariableColumnGrid #Result="{ item: result }" :animation="resultsAnimation">
+                  <article class="result" style="max-width: 300px">
+                    <BaseResultImage class="x-picture&#45;&#45;colored" :result="result">
+                      <template #placeholder>
+                        <div style="padding-top: 100%; background-color: lightgray"></div>
+                      </template>
+                      <template #fallback>
+                        <div style="padding-top: 100%; background-color: lightsalmon"></div>
+                      </template>
+                    </BaseResultImage>
+                    <h1 class="x-title3">{{ result.name }}</h1>
+                  </article>
```

Comment:
add the Banners and Promoteds:

```

                  <template #Banner="{ item: banner }">
                    <Banner :item="banner" />
                  </template>

                  <template #Promoted="{ item: promoted }">
                    <Promoted :item="promoted" />
                  </template>
```

BTW, the props of these components shouldn't be `:banner=""`. and `:promoted=""` ?

Reply:
Done in a different commit so you can revert it if the other reviewer decides to let it as it is.
I'd vote to change it as the entity name is more descriptive.